#                                            Q&A Activity – Week 07

In [32]:
# Import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import ttest_rel
from scipy.stats import ttest_ind
from statsmodels.stats.weightstats import ztest
import warnings
warnings.filterwarnings("ignore")

In [33]:
# Reading the dataset
ReadFile = pd.read_csv('dataset w7.csv')
ReadFile

Gender Test preparation  Total Marks
0     male             none           14
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
6     male             none           30
7   female             none           12
8     male             none           18
9     male             none           24
10    male        completed           44
11    male             none           23
12    male        completed           43
13    male             none           28
14  female             none           41
15  female             none           13
16  female        completed           48
17  female             none           45
18    male             none           47
19    male             none           36
20    male             none           14
21    male        completed           44
22  female             none           41
23  female             none           27
24  female             none           27
25  female             none           38
26  female        completed           50
27  female        completed           43

In [34]:
# Check for Null values
ReadFile.isnull().sum()

Gender              0
Test preparation    0
Total Marks         0
dtype: int64

# 1. Carry out descriptive analytics on the dataset

In [35]:
ReadFile['Gender'].value_counts()

female    16
male      12
Name: Gender, dtype: int64

In [36]:
ReadFile[('Total Marks')].describe()

count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000
Name: Total Marks, dtype: float64

In [37]:
ReadFile[('Test preparation')].describe()

count       28
unique       2
top       none
freq        21
Name: Test preparation, dtype: object

In [38]:
ReadFile.groupby('Gender')['Test preparation'].value_counts()

Gender  Test preparation
female  none                12
        completed            4
male    none                 9
        completed            3
Name: Test preparation, dtype: int64

In [39]:
ReadFile.groupby('Gender')['Total Marks'].mean()

Gender
female    33.750000
male      30.416667
Name: Total Marks, dtype: float64

In [40]:
ReadFile.groupby('Test preparation')['Total Marks'].mean()

Test preparation
completed    44.571429
none         28.238095
Name: Total Marks, dtype: float64

Inferences:

 - There are no null values and the total sample size is 28.
 - There 16 female and 12 male samples.
 - The mean, minimum and maximum score of the total marks are 32.32, 12 and 50 respectively.
 - Out of 28 samples; 21 persons didn't take part in test preparation and the remaining 7 has completed test preparation, out of which 4 are females.
 - The average total marks for females(33.75) are greater than males (30.41).
 - The average total marks for those who completed test preparation (44.57) are greater than for those who didn't take part in test preparation (28.23).

# 2. Check whether female and male students scored the same marks

# Hypothesis:

H0: Female and Male students scored same marks.

Ha: Female and Male students doesn't score same marks.

In [41]:
alpha=.05

FemaleMarks = ReadFile[ReadFile["Gender"]=="female"][["Total Marks"]]
MaleMarks = ReadFile[ReadFile["Gender"] == "male"][["Total Marks"]]

t_score,p_value = ttest_ind(MaleMarks,FemaleMarks)
print('Results are:', 'T-Score:', t_score, '  P Value:', p_value)

Results are: T-Score: [-0.69408853]   P Value: [0.49378223]


Inference:
Since the P-Value (0.49378223) is greater than alpha (0.05), we fail to reject null hypothesis (H0) and reject alternate hypothesis (Ha).

Conclusion: 
Even though the average score of female students is higher than male students, according to hypothesis testing we can conclude that there is no much significant differenece of marks based on gender.

# 3. Whether test preparation helps the students or not.

# Hypothesis:

H0: There is no significant difference between the scores of students who completed test preparation than who did not.

Ha: There is a significant difference between the scores of students who completed test preparation than who did not.

In [31]:
alpha=.05

CmpltdMarks = ReadFile[ReadFile["Test preparation"]=="completed"][["Total Marks"]]
NoneMarks = ReadFile[ReadFile['Test preparation']=="none"][["Total Marks"]]

t_score1,p_value1 = ttest_ind(CmpltdMarks,NoneMarks)
print('Results are:', 'T-Score:', t_score1, '  P Value:', p_value1)

Results are: T-Score: [3.61436375]   P Value: [0.00126713]


Inferences:
Since the P-Value (0.00126713) is less than alpha (0.05), we reject the null hypothesis (H0) and fail to reject the alternate hypothesis (Ha)

Conclusion:
Students who has completed test preparations scored higher than those who did not.
Therefore, we are able to conclude that test preparation significantly helped the students.

                                                -- The End --